src/lwc/geGiftEntryFormApp/geGiftEntryFormApp.js

```diff
@@ -140,19 +141,7 @@ export default class GeGiftEntryFormApp extends NavigationMixin(LightningElement
     batchGiftSubmit(event) {
         const table = this.template.querySelector('c-ge-batch-gift-entry-table');
         this.dataImportRecord = event.detail.dataImportRecord;
-        saveAndDryRunDataImport({batchId: this.recordId, dataImport: this.dataImportRecord})
```

Comment:
This move is because after the dataImportModel comes back from the server it needs to be appended with the additional properties used by columns to display links.  That requires the Data Import object info, which is already in the Table but does not exist in the App.  So either the App needed to get access to the DI object info (by adding another getObjectInfo) or the function needed to move.  I think the latter is better for the structure of our App.  

My opinion on functionality being housed at the App level is evolving ever so slightly. :) Shared state should be managed at the App level, but functionality/logic that is only used by one child component should just be housed in that child component to keep them as loosely coupled as possible and to separate out concerns. This change is a further move toward those two goals.  

The App is still doing an update after loading a row into the form, which could/should also be moved down here to the table, but other than that almost everything concerning displaying/managing the entered gifts (DI rows) is done by the Table component.  None of that logic is shared and there is very little shared state (just the amount and count totals which the Table communicates up to the App via events).  That way the App is working more as an orchestrator - so when a new gift is saved in Batch mode, the App passes the DI to the table which saves the record and then manipulates the response directly in order to display the rows correctly.  Happy to discuss this further or try to add detail to this thought process maybe in our refactoring doc...

Reply:
Thanks - I think that explanation was incredibly clear and catches me up to the latest-and-greatest thinking! Thank you!